pilot/pkg/config/kube/crd/controller/controller.go

```diff
@@ -100,7 +119,11 @@ func NewController(client *Client, options controller2.Options) model.ConfigStor
 
 	// add stores for CRD kinds
 	for _, s := range client.Schemas().All() {
-		out.addInformer(s, options.WatchedNamespace, options.ResyncPeriod)
+		if crdExistsWithRetry(client, s) {
+			out.addInformer(s, options.WatchedNamespace, options.ResyncPeriod)
+		} else {
+			log.Warnf("Skipping CRD %v as it is not present", s.String())
```

Comment:
It would be helpful to be able to query istiod for watched/skipped CRDs. For example, if logs were dropped, automated post-install checks.

Reply:
It might also be helpful to tell the user they need to restart istiod if the missing CRD was added. Maybe something like the following:

"Not watching CRD %v - resource type not found. Restart is required if missing CRD is added"